### ISSUE TITLE
Platform_TimeZoneNames failing for Iran time zones on macOS

### ISSUE THREAD

**Issue by jkoritzinsky** [dotnet/runtime#83901]
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=216592
Build error leg or test failing: System.Tests.TimeZoneInfoTests.Platform_TimeZoneNames
Pull request: https://github.com/dotnet/runtime/pull/83755

## Error Message
Fill the error message using [known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).

```json
{
  "ErrorMessage": "(UTC+03:30) Iran Standard Time (Tehran)",
  "BuildRetry": false,
  "ErrorPattern": "",
  "ExcludeConsoleLog": true
}
```

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

**Comment by jkoritzinsky** [comment]
@hoyosjs Any ideas why this isn't catching the failures? The message is in the test failure:
https://dev.azure.com/dnceng-public/public/_build/results?buildId=216592&view=ms.vss-test-web.build-test-results-tab&runId=4043580&paneView=debug&resultId=172087

**Comment by jkoritzinsky** [comment]
Looks like the text is in the specific subcase that failed. Maybe the Known Issues logic isn't searching in test subcases that failed?

**Comment by tarekgh** [comment]
```
[FAIL] System.Tests.TimeZoneInfoTests.Platform_TimeZoneNames(tzi: (UTC+03:30) Iran Standard Time (Tehran), displayName: "(UTC+03:30) Iran Time", standardName: "Iran Standard Time", daylightName: "Iran Daylight Time")
Assert.Equal() Failure
                                 ↓ (pos 31)
Expected: ···: "(UTC+03:30) Iran Time", StandardName: Iran Standard Time",···
Actual:   ···: "(UTC+03:30) Iran Standard Time (Tehran)", StandardName: Ir···
                                 ↑ (pos 31)
   at System.Tests.TimeZoneInfoTests.Platform_TimeZoneNames(TimeZoneInfo tzi, String displayName, String standardName, String daylightName) + 0x71c
   at System.Runtime!<BaseAddress>+0x11ef4a0
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0x14c
[SKIP] System.Reflection.Tests.ReflectionCacheTests.InvokeClearCache_NoExceptions
[SKIP] System.Reflection.Tests.ReflectionCacheTests.GetMembers_MultipleCalls_ClearCache_ReflectionCacheTestsType
[SKIP] System.Reflection.Tests.ReflectionCacheTests.GetMembers_MultipleCalls_ClearCache_All
Finished System.Runtime.Tests, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
```

**Comment by hoyosjs** [comment]
> Looks like the text is in the specific subcase that failed. Maybe the Known Issues logic isn't searching in test subcases that failed?

@dotnet/dnceng do you know why this blob isn't matching? It seems to fit and it appears in the error section of the test case.

**Comment by carlossanlop** [comment]
This is also happening in 7.0, so I'm reopening the issue.
Can we please backport the fix? Reminder to backport to `release/7.0-staging`.

Affected 7.0 PR is: https://github.com/dotnet/runtime/pull/83747
Affected 6.0 PR is: https://github.com/dotnet/runtime/pull/83748

Failure message in both is:
```
    System.Tests.TimeZoneInfoTests.Platform_TimeZoneNames(tzi: (UTC+03:30) Iran Standard Time (Tehran), displayName: "(UTC+03:30) Iran Time", standardName: "Iran Standard Time", daylightName: "Iran Daylight Time") [FAIL]
      Assert.Equal() Failure
                                       ↓ (pos 31)
      Expected: ···: "(UTC+03:30) Iran Time", StandardName: Iran Standard Time",···
      Actual:   ···: "(UTC+03:30) Iran Standard Time (Tehran)", StandardName: Ir···
                                       ↑ (pos 31)
      Stack Trace:
        /_/src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs(140,0): at System.Tests.TimeZoneInfoTests.Platform_TimeZoneNames(TimeZoneInfo tzi, String displayName, String standardName, String daylightName)
        /_/src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs(370,0): at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

**Comment by Clockwork-Muse** [comment]
... so, why do we have tests that are dependent on external data with no guarantee of stability?  Should we be just checking for a non-empty string?  That it contains part of some content?

**Comment by tarekgh** [comment]
@Clockwork-Muse it is good to be aware of the changes and find out if the change make sense too.

**Comment by carlossanlop** [comment]
> @hoyosjs Any ideas why this isn't catching the failures?

The json pattern is correct, but the `Known Build Error` label was missing. That label is what makes the bot start monitoring failures that match the json pattern.

I added the label a few hrs ago but it isn't matching the 7.0 and 6.0 patterns. I suspect it's because the ExcludeConsoleLog item was set to true. I changed it to false, hopefully that'll make the failures to get collected.
